### PR TITLE
Rocks respawn

### PR DESCRIPTION
**Pull request by antists18**
inserted function spawn_new_rock which checks if new rock can be placed

### CODE REVIEW

File: maps/wasteland/rocks_yield_ore_veins.lua [hunk 1]
```diff
@@ -199,6 +241,16 @@ local function on_player_mined_entity(event)
     end
 end
 
+
+local function on_entity_died(event)
+
+    if not valid_entities[event.entity.name] then
+        return
+    end
+    log("on entity dies works")
```
Comment (nilsh) [inline maps/wasteland/rocks_yield_ore_veins.lua]: pls remove extra log

File: maps/wasteland/rocks_yield_ore_veins.lua [hunk 2]
```diff
@@ -19,6 +25,41 @@ local size_raffle = {
...
+        if x_positive == 2 then
+            pos_x = pos_x * -1
+        end
+        if y_positive == 2 then
+            pos_y = pos_y * -1
+        end
+        position = {x = pos_x, y = pos_y}
+        local rock_type = rock_types[math_random(1, #rock_types)]
+        if surface.can_place_entity({name = rock_type[1], position = position, force = 'neutral'}) then
+            if surface.count_entities_filtered({ position = position, radius = 2.0, force = force_names})<= 0  then
```
Comment (nilsh) [inline maps/wasteland/rocks_yield_ore_veins.lua]: Maybe more radius, 5?

File: maps/wasteland/rocks_yield_ore_veins.lua [hunk 3]
```diff
@@ -11,6 +11,12 @@ local valid_entities = {
     ['sand-rock-big'] = true
 }
 
+local rock_types = {
+    {'rock-big'},
+    {'rock-huge'},
+    {'sand-rock-big'}
+}
```
Comment (nilsh) [inline maps/wasteland/rocks_yield_ore_veins.lua]: Can we reuse the other list?

**Review by nilsh — APPROVED** [review]

**Review by blubFisch — APPROVED** [review]

**Comment by blubFisch** [comment]
THx!

**Pull request merged**